master/buildbot/db/enginestrategy.py

```diff
@@ -209,6 +238,13 @@ def check_sqlalchemy_version(self):
                 raise RuntimeError("SQLAlchemy version %s is not supported by "
                                    "SQLAlchemy-Migrate version %d.%d.%d" % (version, mvt[0], mvt[1], mvt[2]))
 
+    def get_proxy(self, drivername):
```

Comment:
Indeed, after re-reading
https://en.wikipedia.org/wiki/Proxy_pattern
the method itself might be the proxy, but not the returned objects.
What about 
- renaming 'get_proxy' to 'get_strategy', and 
- removing 'Proxy' from the name of the returned objects' classes (ex: MySQLStrategyProxy to MySQLStrategy) ?